### PR TITLE
TE-1036 / 24.04.2 / Adding read-only and share admin boot environments test cases

### PR DESCRIPTION
**Pull request by ericbsd**

### CODE REVIEW

**Comment by ericbsd** [comment]
image

File: test_cases/webui/restricted_admin/read-only-admin/read_only_admin_boot_environments.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        """
+        This test verifies the read-only admin can view the list of boot environments.
+        1. Navigate to Boot Environment page.
+        2. Verify default and created boot environments is visible.
+        """
+        assert Boot.assert_boot_environment_row_exist(shared_config['DEFAULT_BE']) is True
+        assert Boot.assert_boot_environment_row_exist('test-bootenv') is True
+
+    @allure.story("Read Only Admin Can Not Change the Keep Status, Activate, Clone, Delete and Rename Boot Environments")
+    def test_read_only_admin_cannot_to_change_the_keep_status_active_clone_delete_and_rename_boot_environments(self):
```
Comment (markbainix) [inline test_cases/webui/restricted_admin/read-only-admin/read_only_admin_boot_environments.py:51]: suggest:

> def test_read_only_admin_cannot_modify_boot_environments(self):
>         """
>         This test verifies the read-only admin cannot modify boot environments.
>         1. Navigate to Boot Environment page.
>         2. Verify default boot environment is restricted. (keep status, clone rename)
>         3. Verify other boot environment is restricted. (keep status, activate, clone, delete, rename)
>

File: test_cases/webui/restricted_admin/read-only-admin/read_only_admin_boot_environments.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        assert Boot.assert_keep_boot_environment_is_restricted(shared_config['DEFAULT_BE']) is True
+
+        assert Boot.assert_activate_boot_environment_is_restricted('test-bootenv') is True
+        assert Boot.assert_clone_boot_environment_is_restricted('test-bootenv') is True
+        assert Boot.assert_delete_boot_environment_is_restricted('test-bootenv') is True
+        assert Boot.assert_rename_boot_environment_is_restricted('test-bootenv') is True
+        assert Boot.assert_keep_boot_environment_is_restricted('test-bootenv') is True
+
+    @allure.story("Read Only Admin Can Not Scrub the Boot Pool")
+    def test_read_only_admin_cannot_to_scrub_the_boot_pool(self):
```
Comment (markbainix) [inline test_cases/webui/restricted_admin/read-only-admin/read_only_admin_boot_environments.py:69]: remove "to_"

File: test_cases/webui/restricted_admin/read-only-admin/read_only_admin_boot_environments.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        This test verifies the read-only admin can not scrub the boot pool.
+        1. Navigate to Boot Environment page.
+        2. Verify scrub button is restricted.
+        """
+        assert Boot.asser_scrub_boot_environment_button_is_restricted() is True
+
+    @allure.story("Read Only Admin Can Not View the Stats Settings for the Boot Pool")
+    def test_read_only_admin_cannot_view_the_stats_settings_for_the_boot_pool(self):
+        """
+        This test verifies the read-only admin can not view the stats settings for the boot pool.
```
Comment (markbainix) [inline test_cases/webui/restricted_admin/read-only-admin/read_only_admin_boot_environments.py:80]: "can not" -> "cannot"

File: keywords/webui/boot.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        """
+        This method returns True if the boot pool status is active.
+
+        :param status: The status of the boot pool.
+        :return: True if the boot pool status is active otherwise it returns False.
+
+        Example:
+            - Boot.assert_boot_pool_status('active')
+        """
+        return WebUI.wait_until_visible(xpaths.boot.boot_pool_status(status))
```
[the remark this answers is not in the document]
Comment (ericbsd) [reply keywords/webui/boot.py:33]: The status is upper case, not lower case.

**Comment by ericbsd** [comment]
image

File: keywords/webui/boot.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        :return: True if the boot environment exists in the list otherwise it returns False.
+
+        Example:
+            - Boot.assert_boot_environment_row_exist('test-bootenv')
+        """
+        be_xpath = COM.convert_to_tag_format(be_name)
+        return WebUI.wait_until_visible(f'//*[@data-test="row-{be_xpath}"]')
+
+    @classmethod
+    def assert_boot_environment_element_restricted(cls, be_name, link_action) -> bool:
```
Comment (markbainix) [inline keywords/webui/boot.py:42]: parameters do not have types `: str`

**Review by markbainix — APPROVED** [review]

**Pull request merged**